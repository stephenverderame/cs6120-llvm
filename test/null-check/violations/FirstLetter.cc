#include <cassert>
#include <cstdio>
#include <cstdlib>
// NOLINTBEGIN
int main(int argc, const char** argv)
{
    if (argc > 256) {
        printf("Too many arguments\n");
        return 1;
    }
    int arg_num = argc - 1;
    char args[257];
    for (int i = 0; i < arg_num; ++i) {
        const auto arg = argv + i + 1;
        if (arg) {
            const auto word = *arg;
            if (word) {
                args[i] = *word;
            }
        }
    }

    args[arg_num] = '\0';

    printf("%s\n", args);
}
// NOLINTEND