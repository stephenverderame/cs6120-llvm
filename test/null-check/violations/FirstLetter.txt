[31mSafety Violation[00m: Use of potentially null pointer in store at FirstLetter.cc:23
Because an index could not be proven to be in range, the following counterexample was found:
	[1marg_num[0m = (- 1)
	[1margc[0m = 0
	[1mi[0m = 1
