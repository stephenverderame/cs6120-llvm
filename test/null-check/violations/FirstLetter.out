[31mSafety Violation[00m: Use of potentially null pointer in store at FirstLetter.cc:23
